### PR TITLE
New KPI endpoint for Access requests data

### PR DESCRIPTION
**Pull request by Stephan-Hough**
feat/water2644
--KPI route and handler added to return access requests data

### CODE REVIEW

File: test/modules/kpi-reporting/controller.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  experiment('when data is returned from the repo', async () => {
+    beforeEach(async => {
+      sandbox.stub(repos, 'getEntityRolesKPIdata').resolves(repoData);
+    });
+    test('the right data shape is returned', async () => {
+      const { data } = await controller.getKPIEntityRolesData();
+      expect(data.totals.allTime).to.equal(130);
+      expect(data.totals.ytd).to.equal(30);
+      expect(data.monthly).to.be.array();
+      expect(Object.keys(data.monthly[0])).to.equal(['month', 'total', 'change']);
```
Comment (DGajewska) [inline test/modules/kpi-reporting/controller.js:52]: I'm not sure that these are required as they are confirmed by the 3 tests below

Comment (Stephan-Hough) [reply test/modules/kpi-reporting/controller.js]: Thanks - removed that line.

File: src/lib/repo/kpi-reporting.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+'use-strict';
+const { pool } = require('../connectors/db');
+
+/**
+ * returns the number of delegated access granted by month for the current year including the % change by month
```
Comment (james-siteclick) [inline src/lib/repo/kpi-reporting.js:5]: Not sure the comment is accurate as I don't think the % change is included

File: src/modules/kpi-reporting/controller.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+'user-strict';
+const repo = require('../../lib/repo/kpi-reporting');
+const Boom = require('@hapi/boom');
+
+const months = ['January', 'February', 'March', 'April', 'May', 'June',
+  'July', 'August', 'September', 'October', 'November', 'December'];
+
+const percentageChage = (index, data, row) => {
+  if (index < (data.length - 1)) {
+    return (row.total - data[(index + 1)].total) / (data[(index + 1)].total < 1 ? 1 : data[(index + 1)].total) * 100;
```
Comment (james-siteclick) [inline src/modules/kpi-reporting/controller.js:10]: I'm not sure we should be handling the /0 issue in this way.  I'm not sure it is mathematically correct.
I wonder if we should do along the lines of:
```
const numerator = row.total - data[(index + 1)].total;
const denominator = data[(index + 1)].total;
if(numerator === 0 && denominator === 0) {
  return 0;
}
const result = numerator/denominator;
return result.isNaN() ? '∞' : result * 100;
```

Comment (Stephan-Hough) [reply src/modules/kpi-reporting/controller.js]: done thanks

**Pull request merged**